src/core/Akka.Streams.Tests/Dsl/FramingSpec.cs

```diff
@@ -240,43 +241,98 @@ private static string RandomString(int length)
+            ByteString payload, 
+            int fieldLength, 
+            ByteOrder byteOrder, 
+            ByteString offset, 
+            ByteString tail)
         {
             var h = ByteString.FromBytes(new byte[4].PutInt(payload.Count, order: byteOrder));
             var header = byteOrder == ByteOrder.LittleEndian ? h.Slice(0, fieldLength) : h.Slice(4 - fieldLength);
 
-            return ByteString.FromBytes(new byte[fieldOffset]) + header + payload;
+            return offset + header + payload + tail;
         }
 
         [Fact]
         public void Length_field_based_framing_must_work_with_various_byte_orders_frame_lengths_and_offsets()
         {
-            var counter = 1;
-            foreach (var byteOrder in ByteOrders)
+            IEnumerable<Task<(IEnumerable<ByteString>, List<ByteString>, (ByteOrder, int, int))>> GetFutureResults()
             {
+                foreach (var byteOrder in ByteOrders)
                 foreach (var fieldOffset in FieldOffsets)
+                foreach (var fieldLength in FieldLengths)
                 {
-                    foreach (var fieldLength in FieldLengths)
+                    var encodedFrames = FrameLengths.Where(x => x < 1L << (fieldLength * 8)).Select(length =>
                     {
-                        var encodedFrames = FrameLengths.Where(x => x < 1L << (fieldLength * 8)).Select(length =>
-                          {
-                              var payload = ReferenceChunk.Slice(0, length);
-                              return Encode(payload, fieldOffset, fieldLength, byteOrder);
-                          }).ToList();
-
-                        var task = Source.From(encodedFrames)
-                            .Via(Rechunk)
-                            .Via(Framing.LengthField(fieldLength, int.MaxValue, fieldOffset, byteOrder))
-                            .Grouped(10000)
-                            .RunWith(Sink.First<IEnumerable<ByteString>>(), Materializer);
-
-                        task.Wait(TimeSpan.FromSeconds(3)).Should().BeTrue();
-                        task.Result.Should().BeEquivalentTo(encodedFrames);
-
-                        _helper.WriteLine($"{counter++} from 80 passed");
+                        var payload = ReferenceChunk.Slice(0, length);
+                        return Encode(payload, fieldOffset, fieldLength, byteOrder);
+                    }).ToList();
+
+                    yield return Source.From(encodedFrames)
+                        .Via(Rechunk)
+                        .Via(Framing.LengthField(fieldLength, int.MaxValue, fieldOffset, byteOrder))
+                        .Grouped(10000)
+                        .RunWith(Sink.First<IEnumerable<ByteString>>(), Materializer)
+                        .ContinueWith(t => (t.Result, encodedFrames, (byteOrder, fieldOffset, fieldLength)));
+                }
+            }
+
+            Parallel.ForEach(GetFutureResults(), async futureResult => 
+            { 
```

Comment:
Check this out, because introducing parallelism in tests that do not require to execute their asserts in order could save a lot of time. 

This commit brought the execution time of each of both tests from `1.2 min` to less than `2s`

Reply:
Oh wow, I wonder where else we could take advantage of this.